terraform.tf.j2

```diff
@@ -130,6 +144,16 @@ provider "aws" {
   }
 }
 
+provider "aws" {
+  version = "~> 3.12.0"
```

Comment:
Don't need the version here as it's already in the `required_providers` at the top; less places to try and keep in sync.